plugins/optimization-detective/class-od-url-metrics-group-collection.php

```diff
@@ -217,12 +248,21 @@ public function get_group_for_viewport_width( int $viewport_width ): OD_URL_Metr
 	 * @return bool Whether all groups have some URL metrics.
 	 */
 	public function is_every_group_populated(): bool {
-		foreach ( $this->groups as $group ) {
-			if ( count( $group ) === 0 ) {
-				return false;
-			}
+		if ( array_key_exists( __FUNCTION__, $this->result_cache ) ) {
+			return $this->result_cache[ __FUNCTION__ ];
 		}
```

Comment:
I'm not entirely happy with this. There's too much boilerplate. I wanted to instead have a `get_cached_result()` helper method that relied on generics for PHPStan strict type checks. However, I wasn't able to get it to work. See https://phpstan.org/r/b203ec8c-2d70-469c-b24f-9e7143cff72d

Reply:
Granted, it would also have to be updated to support caching by the method argument as in the `get_group_for_viewport_width` example above.

Reply:
I've reported this as a possible bug to PHPStorm: https://github.com/phpstan/phpstan/issues/11138

Reply:
Hmm I thought I had a similar issue recently but now I can't find the code anymore 😅  Curious to hear the response on that issue 👍